codegenerator/cli/templates/static/codegen/src/globalState/GlobalStateManager.res

```diff
@@ -35,11 +35,14 @@ module MakeManager = (S: State) => {
       NodeJsLocal.process->NodeJsLocal.exitWithCode(Failure)
     }
   }
-  and dispatchTask = (self, task: S.task) => Js.Global.setTimeout(() => {
+  and dispatchTask = (self, task: S.task) => {
+    let stateId = self.state->S.getId
+    Js.Global.setTimeout(() => {
       S.taskReducer(self.state, task, ~dispatchAction=action =>
-        dispatchAction(~stateId=self.state->S.getId, self, action)
+        dispatchAction(~stateId, self, action)
       )->ignore
     }, 0)->ignore
+  }
```

Comment:
This was a bug, the state id should be retrieved at the time dispatchTask gets called, not as dispatchAction gets called.

Since S.getId was being called only at dispatchAction it would always pass the latest stateId and there would be no way to invalidate the action against it's state id